go/test/endtoend/cluster/vttablet_process.go

```diff
@@ -149,8 +149,9 @@ func (vttablet *VttabletProcess) Setup() (err error) {
 		}
 	}()
 
+	time.Sleep(10 * time.Second)
 	if vttablet.ServingStatus != "" {
-		if err = vttablet.WaitForTabletStatus(vttablet.ServingStatus); err != nil {
+		if err = vttablet.WaitForTabletStatus(vttablet.ServingStatus, "SERVING"); err != nil {
```

Comment:
From offline conversation I recall that the expected status here is `NOT_SERVING` and the test was stuck waiting for it because the tablet has already transitioned to `SERVING`. That seems like a deeper issue that this change tries to paper over.
Why is the expected status `NOT_SERVING`? Is that correct in the first place?

Reply:
I think the purpose of `vttablet.WaitForTabletStatus` is to ensure the vttablet started properly, its health data is available, and that no error were returned by the vttablet process. We don't actually care much which status we are returning as long as the process is healthy, checking for both `SERVING` and `NOT_SERVING` is okay.

Running the test on a slow machine might lead vttablet to answer `NOT_SERVING`, whereas a faster machine would maybe return `SERVING`.

I cleaned the code a little bit in b8ef6962bb.